firmware/hw_layer/debounce.cpp

```diff
@@ -0,0 +1,43 @@
+/**
+ * @file        debounce.cpp
+ * @brief       Generic button debounce class
+ *
+ * @date Aug 31, 2020
+ * @author David Holdeman, (c) 2020
+ */
+#include "debounce.h"
+
+void ButtonDebounce::init (int t, brain_pin_e p, pin_input_mode_e m) {
+    threshold = t;
+    timeLast = 0;
+    pin = p;
+#ifdef PAL_MODE_INPUT_PULLDOWN
+    // getInputMode converts from pin_input_mode_e to iomode_t
+    mode = getInputMode(m);
+    efiSetPadMode("Button", p, mode);
+#endif
+}
+
+bool ButtonDebounce::readPinEvent() {
+    efitimems_t timeNow = currentTimeMillis();
```

Comment:
it's cheaper to operate in units of `efitick_t`, and convert the timeout period once upon initialization, instead of every time we have to check the time

Reply:
what's the conversion? example?

Reply:
And how do I get the current time in efitick_t?

Reply:
`getTimeNowNt()` will get you the current time in units of `efitick_t`.

`MS2NT(...)` will convert from milliseconds to efitick_t, so you can convert the passed in threshold to use ticks internally

Reply:
getTimeNowNt()